### PR TITLE
Set block event

### PR DESCRIPTION
**Pull request by 2xsaiko**
what can I say, it works

also yeah this is for setBlockState

### CODE REVIEW

**Comment by Devan-Kerman** [comment]
ok but needs more reflective event listeners and overhead

**Comment by 2xsaiko** [comment]
I thought I made up for that by overengineering the shit out of it

File: src/main/java/io/github/fabriccommunity/everything/api/event/level/SetBlockEvent.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        }
+
+        @Override
+        public void setFlags(int flags) {
+            this.flags = flags;
+        }
+
+        @Override
+        public boolean isCancelled() {
+            return cancelled;
```
Comment (2xsaiko) [inline src/main/java/io/github/fabriccommunity/everything/api/event/level/SetBlockEvent.java:231]: very good code

Comment (vini2003) [reply src/main/java/io/github/fabriccommunity/everything/api/event/level/SetBlockEvent.java:231]: i mean
it works

**Comment by Juuxel** [comment]
Does this have methods for querying the ages of different handlers from an AgeDatabase:tm:? I need to be sure my handlers are adults, i.e. over 18 ms old.

**Comment by 2xsaiko** [comment]
It has implicit support

File: src/main/java/io/github/fabriccommunity/everything/api/event/level/SetBlockEvent.java (hunk 1 above)
Comment (valoeghese) [reply src/main/java/io/github/fabriccommunity/everything/api/event/level/SetBlockEvent.java:231]: epic

**Pull request merged**